src/Service/ImageHandler.php

```diff
@@ -41,27 +44,42 @@ public function fetch(string $url): string
         }
 
         $response = $this->client->request('GET', $url);
-        if (200 !== $response->getStatusCode()) {
+        if (Response::HTTP_OK !== $response->getStatusCode()) {
             throw new ImageFetchException(sprintf('Failed to fetch %s with code %s', $url, $response->getStatusCode()), $response->getStatusCode());
         }
 
-        $dest = $path.basename($url);
-        $fileHandler = fopen($dest, 'w');
-        foreach ($this->client->stream($response) as $chunk) {
-            fwrite($fileHandler, $chunk->getContent());
+        $headers = $response->getHeaders();
+        $type = reset($headers['content-type']);
+        $dest = $path.$this->generateLocalFilename($url, $type);
```

Comment:
`$type` may be `false` here (https://www.php.net/manual/en/function.reset.php#refsect1-function.reset-returnvalues). Doesn't Psalm complain about this?